pkg/pool-weighted/contracts/smart/ManagedPool.sol

```diff
@@ -223,6 +237,62 @@ contract ManagedPool is BaseWeightedPool, ProtocolFeeCache, ReentrancyGuard {
         return _managementSwapFeePercentage;
     }
 
+    function getSwapFeePercentage() public view virtual override returns (uint256) {
+        // Load current pool state from storage
+        bytes32 poolState = _getMiscData();
+
+        uint256 startTime = poolState.decodeUint32(_FEE_START_TIME_OFFSET);
+        uint256 endTime = poolState.decodeUint32(_FEE_END_TIME_OFFSET);
+
+        return
+            GradualValueChange.getInterpolatedValue(
+                super.getSwapFeePercentage(),
+                _endSwapFeePercentage,
+                startTime,
+                endTime
+            );
+    }
+
+    /**
+     * @dev Return start time, end time, and endWeights as an array.
+     * Current weights should be retrieved via `getNormalizedWeights()`.
+     */
+    function getGradualFeeUpdateParams()
+        external
+        view
+        returns (
+            uint256 startTime,
+            uint256 endTime,
+            uint256 endSwapFeePercentage
```

Comment:
I think we should include the `startSwapFeePercentage` here as otherwise you can't calculate the current swap fee from the return value from this function.

Reply:
Yes; sorry - was getting late :)